store/mongo/datastore_mongo.go

```diff
@@ -981,7 +993,7 @@ func (db *DataStoreMongo) SearchDevices(ctx context.Context, searchParams model.
 		return nil, -1, errors.Wrap(err, "failed to search devices")
 	}
 
-	count, err := c.CountDocuments(ctx, findQuery)
+	count, err := c.CountDocuments(ctx, filterWithTenantID)
 	if err != nil {
 		return nil, -1, errors.Wrap(err, "failed to search devices")
 	}
```

Comment:
! missing `tenant_id` filter on ListGroups `Distinct` operation.

Reply:
! missing match stage to filter on `tenant_id` in `GetAllAttributeNames` aggregation pipeline.